src/OpenDiffix.Core/Value.fs

```diff
@@ -46,6 +47,20 @@ let comparer direction nulls =
     | Null, Null -> 0
     | Null, _ -> nullsValue
     | _, Null -> -nullsValue
+    | String x, String y ->
+      // Using PostgreSQL string comparison as a template
+      let compareInfo = CompareInfo.GetCompareInfo("en-US")
+      // we want whitespace & punctuation comparison ("symbols" in .NET) to have smaller priority,
+      // so we ignore first.
+      let comparisonIgnoreSymbols =
+        directionValue
+        * compareInfo.Compare(x, y, CompareOptions.IgnoreSymbols + CompareOptions.StringSort)
+      // if the former gives a tie, we include symbols.  `StringSort` means they come last and we group letter cases together
+      // reference: https://wiki.postgresql.org/wiki/FAQ#Why_do_my_strings_sort_incorrectly.3F
+      if (comparisonIgnoreSymbols <> 0) then
+        comparisonIgnoreSymbols
+      else
+        directionValue * compareInfo.Compare(x, y, CompareOptions.StringSort)
```

Comment:
Can we memo this so that it's outside the lambda? In the callsite I'd then memo the func instead of creating a new one for each sorting pair (may be more expensive).